### ISSUE TITLE
[vLLM] Implement prompt_logprobs support

### ISSUE THREAD

**Issue by kmabeeTT** [tenstorrent/tt-xla#3651]
## Problem

`prompt_logprobs` is plumbed through the TT plugin but never computed. `model_runner.py:1351` hardcodes every entry to `None`:

```python
prompt_logprobs_dict: dict[str, Optional[LogprobsTensors]] = {}
for req_id in self.input_batch.req_ids[:num_reqs]:
    prompt_logprobs_dict[req_id] = None
```

The `num_prompt_logprobs` value is correctly tracked per request (`model_runner.py:576-580`) and stored in `input_batch.py:339`, but never used to compute actual logprobs on prompt tokens.

## What's needed

- During the prefill pass, compute `log_softmax` on prompt token logits and gather top-k (similar to existing `gather_logprobs` for generated tokens)
- Populate `prompt_logprobs_dict` with real `LogprobsTensors` instead of `None`
- Add E2E test verifying prompt token logprobs are returned and have correct structure/values
- Consider whether this can reuse the existing `gather_logprobs` compiled graph or needs a separate graph variant (prompt tokens may have different batch dimensions)
